base/loading.jl

```diff
@@ -415,14 +415,15 @@ end
 evalfile(path::AbstractString, args::Vector) = evalfile(path, UTF8String[args...])
 
 function create_expr_cache(input::AbstractString, output::AbstractString)
-    isfile(output) && rm(output)
```

Comment:
possibly, or maybe `rm` should be done in the `catch` below? let's see what the tests for #12720 say with the current version on CI